File: backend/src/validators/insufficient_limit.py
from ..core.state import Store


def validate(store: Store, transaction_data: dict) -> list[str]:
    account = store.get_account()
    if account is None:
        return []
    amount = transaction_data.get("amount", 0)
    if account.available_limit < amount:
        return ["insufficient-limit"]
    return []

File: frontend/src/core/validators/insufficientLimit.js
export function validate(store, transactionData) {
  const account = store.getAccount();
  if (account === null) {
    return [];
  }
  const amount = transactionData.amount || 0;
  if (account.availableLimit < amount) {
    return ['insufficient-limit'];
  }
  return [];
}